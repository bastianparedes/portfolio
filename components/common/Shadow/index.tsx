import React from 'react';
import styles from './styles.module.scss';
import classNames from '../../../utils/classNames';

interface ShadowProps {
  children?: React.ReactNode;
  setVisible?: ((boolean: boolean) => void) | null;
  modes?: string[];
}

const Shadow = ({
  children,
  setVisible = null,
  modes = ['dark']
}: ShadowProps): JSX.Element => {
  const handleOnClick = (event: React.MouseEvent): void => {
    event.preventDefault();
    event.stopPropagation();

    if (setVisible === null) return;

    const target = event.target as HTMLElement;
    const classList = target.classList;
    const shadoWasClicked = classList.contains(styles.shadow);

    if (shadoWasClicked) setVisible(false);
  };

  const classNamesList = modes.map((mode) => styles[mode]);

  return (
    <div
      data-testid="data-testid-shadow"
      className={classNames(styles.shadow, ...classNamesList)}
      onClick={handleOnClick}
    >
      {children}
    </div>
  );
};

export default Shadow;
