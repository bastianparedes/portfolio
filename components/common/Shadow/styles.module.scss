@import 'styles/_constants';

.shadow {
  width: 100vw;
  height: 100vh;
  position: fixed;
  left: 0;
  top: 0;
  z-index: $z-index-shadow;
}

.dark {
  background-color: rgb(0, 0, 0, 0.5);
}

.blur {
  -webkit-backdrop-filter: blur(5px);
  backdrop-filter: blur(5px);
}
