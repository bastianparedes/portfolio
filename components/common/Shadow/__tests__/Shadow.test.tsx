import React from 'react';
import Shadow from '..';
import { fireEvent, render } from '@testing-library/react';

describe('<Shadow />', () => {
  it('should render', () => {
    const props = {
      setVisible: jest.fn()
    };

    const { container } = render(<Shadow {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without props', () => {
    const { container, getByTestId } = render(<Shadow />);
    fireEvent.click(getByTestId('data-testid-shadow'));
    expect(container).toMatchSnapshot();
  });

  it('should call setVisible when shadow is clicked', () => {
    const props = {
      setVisible: jest.fn()
    };

    const { getByTestId } = render(<Shadow {...props} />);
    fireEvent.click(getByTestId('data-testid-shadow'));

    expect(props.setVisible).toBeCalledTimes(1);
    expect(props.setVisible).toBeCalledWith(false);
  });
});
