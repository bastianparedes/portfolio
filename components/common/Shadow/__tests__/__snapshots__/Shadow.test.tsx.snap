// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shadow /> should render 1`] = `
<div>
  <div
    class="shadow dark"
    data-testid="data-testid-shadow"
  />
</div>
`;

exports[`<Shadow /> should render without props 1`] = `
<div>
  <div
    class="shadow dark"
    data-testid="data-testid-shadow"
  />
</div>
`;
